api/base/exceptions.py

```diff
@@ -36,11 +46,27 @@ def json_api_exception_handler(exc, context):
     return response
 
 
+class JSONAPIException(APIException):
```

Comment:
@brianjgeiger the current logic here feels a bit brittle, so I'm adding a new class as an alternative.

I'd like to avoid feature creep on this PR, but am proposing the following: Ignore all exceptions that use the old pattern for now. I see a lot of places where we raise exceptions sans useful information, and think probably an exception message audit may be beneficial.